src/TpetraLinearSystem.C

```diff
@@ -1474,64 +1492,50 @@ void TpetraLinearSystem::applyDirichletBCs(stk::mesh::FieldBase * solutionField,
     & stk::mesh::selectField(*solutionField)
     & !(realm_.get_inactive_selector());
 
-  stk::mesh::BucketVector const& buckets =
-    realm_.get_buckets( stk::topology::NODE_RANK, selector );
-
-  const bool internalMatrixIsSorted = true;
-  int nbc=0;
-  for(const stk::mesh::Bucket* bptr : buckets) {
-    const stk::mesh::Bucket & b = *bptr;
+  using Traits = nalu_ngp::NGPMeshTraits<>;
+  using MeshIndex = typename Traits::MeshIndex;
 
-    const unsigned fieldSize = field_bytes_per_entity(*solutionField, b) / sizeof(double);
-    ThrowRequire(fieldSize == numDof_);
+  ngp::Mesh ngpMesh = realm_.ngp_mesh();
+  NGPDoubleFieldType ngpSolutionField = realm_.ngp_field_manager().get_field<double>(solutionField->mesh_meta_data_ordinal());
+  NGPDoubleFieldType ngpBCValuesField = realm_.ngp_field_manager().get_field<double>(bcValuesField->mesh_meta_data_ordinal());
 
-    const stk::mesh::Bucket::size_type length   = b.size();
-    const double * solution = (double*)stk::mesh::field_data(*solutionField, *b.begin());
-    const double * bcValues = (double*)stk::mesh::field_data(*bcValuesField, *b.begin());
+  ngpSolutionField.sync_to_device();
+  ngpBCValuesField.sync_to_device();
```

Comment:
Is this just to be safe?

Reply:
Yes. I just didn't want to count on the fields already being sync'd. If they are already sync'd, these sync_to_device calls should be very fast.